homeassistant/components/mqtt/siren.py

```diff
@@ -386,8 +397,9 @@ async def async_turn_off(self, **kwargs: Any) -> None:
             self._attr_is_on = False
             self.async_write_ha_state()
 
-    def _update(self, data: dict[str, Any]) -> None:
+    def _update(self, data: SirenTurnOnServiceParameters) -> None:
         """Update the extra siren state attributes."""
+        update: dict[str, Any] = dict(data)
         for attribute, support in SUPPORTED_ATTRIBUTES.items():
-            if self._supported_features & support and attribute in data:
-                self._attr_extra_state_attributes[attribute] = data[attribute]
+            if self._attr_supported_features & support and attribute in update:
+                self._attr_extra_state_attributes[attribute] = update[attribute]
```

Comment:
IMO, you shouldn't need to create a dict from a TypeDict here.
I would either do a cast, or consider it a failure from mypy and add a type ignore: `# type: ignore[literal-required]`

I would appreciate having the opinion of @cdce8p on this one.

Reply:
`type: ignore` is fine. I've added it for other instances as well in the past. It's often difficult to get full literals for TypedDicts, especially when iterating of dict keys. However, I don't think it's that big of an issue to disable the error-code completely.